core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -369,6 +369,13 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
             channelReadComplete(ctx);
             return;
         }
+        if (evt instanceof InitiateConnectionShutdown) {
+            // HTTP/1 doesn't support draining that signals clients about connection shutdown but still
+            // accepts in flight requests. Simply destroy KeepAliveHandler which causes next response
+            // to have a "Connection: close" header and connection to be closed after the next response.
+            destroyKeepAliveHandler();
+            writer.initiateConnectionShutdown();
```

Comment:
Add `return` unless we need to propagate the `InitiateConnectionShutdown` event?

Reply:
Forgot to add it based on the initial comment, thanks for the reminder!